build-libs/redirects.js

```diff
@@ -155,85 +175,25 @@ async function buildDotIoRedirects() {
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Waypoint repo
-	// TODO: intent is to do this after all products have been migrated
-	const vaultIoRedirects = [...vaultAuthorRedirects]
-
-	const rawVagrantRedirects = isContentDeployPreview('vagrant')
-		? fs.readFileSync(path.join(process.cwd(), '../redirects.js'), 'utf-8')
-		: isDeployPreview()
-		? '[]'
-		: await fetchGithubFile({
-				owner: 'hashicorp',
-				repo: 'vagrant',
-				path: 'website/redirects.js',
-				ref: 'stable-website',
-		  })
-	const vagrantAuthorRedirects = eval(rawVagrantRedirects)
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Vagrant repo
-	// TODO: intent is to do this after all products have been migrated
-	const vagrantIoRedirects = [...vagrantAuthorRedirects]
-
-	const rawPackerRedirects = isContentDeployPreview('packer')
-		? fs.readFileSync(path.join(process.cwd(), '../redirects.js'), 'utf-8')
-		: isDeployPreview()
-		? '[]'
-		: await fetchGithubFile({
-				owner: 'hashicorp',
-				repo: 'packer',
-				path: 'website/redirects.js',
-				ref: 'stable-website',
-		  })
-	const packerAuthorRedirects = eval(rawPackerRedirects)
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Packer repo
-	// TODO: intent is to do this after all products have been migrated
-	const packerIoRedirects = [...packerAuthorRedirects]
-
-	const rawConsulRedirects = isContentDeployPreview('consul')
-		? fs.readFileSync(path.join(process.cwd(), '../redirects.js'), 'utf-8')
-		: isDeployPreview()
-		? '[]'
-		: await fetchGithubFile({
-				owner: 'hashicorp',
-				repo: 'consul',
-				path: 'website/redirects.js',
-				ref: 'stable-website',
-		  })
-	const consulAuthorRedirects = eval(rawConsulRedirects)
-	// TODO: split non-author redirects into dev-portal,
-	// TODO: rather than leaving all redirects in the Packer repo
-	// TODO: intent is to do this after all products have been migrated
-	const consulIoRedirects = [...consulAuthorRedirects]
-	// TODO ... consolidate redirects for other products
 	return [
 		...devPortalToDotIoRedirects,
-		...addHostCondition(boundaryIoRedirects, 'boundary'),
-		...addHostCondition(nomadIoRedirects, 'nomad'),
-		...addHostCondition(sentinelIoRedirects, 'sentinel'),
-		...addHostCondition(vaultIoRedirects, 'vault'),
-		...addHostCondition(waypointIoRedirects, 'waypoint'),
-		...addHostCondition(vagrantIoRedirects, 'vagrant'),
-		...addHostCondition(packerIoRedirects, 'packer'),
-		...addHostCondition(consulIoRedirects, 'consul'),
+		...(await getRedirectsForProduct('boundary')),
+		...(await getRedirectsForProduct('nomad')),
+		...(await getRedirectsForProduct('vault')),
+		...(await getRedirectsForProduct('waypoint')),
+		...(await getRedirectsForProduct('vagrant')),
+		...(await getRedirectsForProduct('packer')),
+		...(await getRedirectsForProduct('consul')),
```

Comment:
Thoughts for parallelizing these async fns?

It could shave off a 10-100 ms off build time.

Reply:
Sure sure, updated!